src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -229,14 +230,14 @@ protected sealed override void InitializeWorker(AnalysisContext context)
             PooledHashSet<string> compilerDiagnosticIds,
             ImmutableArray<string> userExclusions)
         {
-            if (!root.ContainsDirectives)
+            if (!ContainsPragmaDirective(root))
             {
                 return false;
             }
 
             using var _ = ArrayBuilder<string>.GetInstance(out var idsBuilder);
             var hasPragmaInAnalysisSpan = false;
-            foreach (var trivia in root.DescendantTrivia())
+            foreach (var trivia in root.DescendantTrivia(node => node.ContainsDirectives))
```

Comment:
almost definitely .  it can be `node => node.ContainsDirectives && node.IntersectsWith(span)`

Reply:
I believe it will be a more involved change, as the caller expects to have collected pragmas and their IDs for the entire file if there were to be a pragma on this line (it needs to analyze the enable/disable state at the line). So, likely needs a 2 pass approach here - first only operate on directives on the span to return the `hasPragmaDirectivesInAnalysisSpan` question for the caller. And then the caller makes another call without a span to compute state information for all pragmas in the file.

Reply:
yeah, def leave things simple for now.  In the vast majority of files we'll bail out near instantly.  So the win is already there.  It's ok for the rare case to still be as expensive as today.